back/app/services/projects_finder_admin_service.rb

```diff
@@ -234,4 +235,30 @@ def self.filter_current_phase_participation_method(scope, params = {})
 
     scope.where(id: project_ids_with_matching_phase)
   end
+
+  # Filter projects by visibility and listed status
+  def self.filter_visibility(scope, params = {})
+    visibility_params = Array(params[:visibility])
+    return scope if visibility_params.blank?
+
+    valid_visibilities = %w[public groups admins]
+    selected_visibilities = visibility_params & valid_visibilities
+
+    conditions = scope.none
+
+    if selected_visibilities.present?
+      # This generates: WHERE "projects"."visible_to" IN ('public', 'groups', ...)
+      conditions = conditions.or(scope.where(visible_to: selected_visibilities))
+    end
+
+    if visibility_params.include?('unlisted')
```

Comment:
I find it conceptually not very logical that this is included in the visibility. It also makes the code very confusing. 

I think it would be better to not mix concepts and just keep 'visibility' as the visibility access rights (`public, groups, admin`). Then we can add another filter for 'discoverability' which can be `'listed'` or `'unlisted'`